feat: add mongo_date functions to the rule engine (#10408)
emqx/emqx: changes/ee/feat-10408.en.md

> The rule engine SQL-like language has got three more built-in functions for creating values of the MongoDB date type. These functions are only useful for rules with at least one MongoDB bridge action.

Reviewer: I have the impression that those functions when there are _only_ mongo bridge actions (not tested). (just to adjust the message here, not that there is much to do otherwise)

If the same rule using these functions has another bridge as an action, I wonder how the tuples are treated by those.  :thinking:

Reviewer: I think that we should only support the tuples from `mongo_date` in mongo-related actions, indeed.  Hopefully the function name hints the user to avoid using it in other actions.  :joy_cat:

> The rule engine SQL-like language has got three more built-in functions for creating values of the MongoDB date type. These functions are only useful for rules with at least one MongoDB bridge action.

Reviewer: ```suggestion
The rule engine SQL-like language has got three more built-in functions for creating values of the MongoDB date type. These functions are useful for rules with MongoDB bridge actions only and not supported in other actions.
```
